### PR TITLE
[optimize] optimize get_json_xx function

### PR DESCRIPTION
**Pull request by e0c9**
## Proposed changes

Avoid repeated parsing of json string.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — COMMENTED** [review]
The BE unit test failed.
And if you change the signature of functions, it may cause incompatible issue.

**Comment by e0c9** [comment]
> The BE unit test failed. And if you change the signature of functions, it may cause incompatible issue.

Fixed

**Comment by morningman** [comment]
> And if you change the signature of functions, it may cause incompatible issue.

hi @e0c9 
What about this problem?

**Comment by e0c9** [comment]
> > And if you change the signature of functions, it may cause incompatible issue.
> 
> hi @e0c9 What about this problem?

I modified it locally based on the old version. json_string's type has changed. `const std::string& json_string` ->
`const std::string_view& json_string`

File: be/src/exprs/json_functions.cpp [hunk 1]
```diff
@@ -412,41 +419,47 @@ rapidjson::Value* JsonFunctions::get_json_object_from_parsed_json(
...
                                       doris_udf::FunctionContext::FunctionStateScope scope) {
     if (scope != FunctionContext::FRAGMENT_LOCAL) {
         return;
     }
 
-    if (!context->is_arg_constant(1)) {
+    if (!context->is_arg_constant(0) && !context->is_arg_constant(1)) {
         return;
     }
+    StringVal* json_str = reinterpret_cast<StringVal*>(context->get_constant_arg(0));
```
Comment (morningman) [inline be/src/exprs/json_functions.cpp:431]: The first argument may NOT be constant, so `get_constant_arg` may return nullptr.

File: gensrc/script/doris_builtins_functions.py [hunk 2]
```diff
@@ -1096,33 +1096,33 @@
     # Json functions
     [['get_json_int'], 'INT', ['VARCHAR', 'VARCHAR'],
         '_ZN5doris13JsonFunctions12get_json_intEPN9doris_udf15FunctionContextERKNS1_9StringValES6_',
-        '_ZN5doris13JsonFunctions17json_path_prepareEPN9doris_udf15FunctionContextENS2_18FunctionStateScopeE',
-        '_ZN5doris13JsonFunctions15json_path_closeEPN9doris_udf15FunctionContextENS2_18FunctionStateScopeE',
+        '_ZN5doris13JsonFunctions21json_function_prepareEPN9doris_udf15FunctionContextENS2_18FunctionStateScopeE',
```
Comment (morningman) [inline gensrc/script/doris_builtins_functions.py:1099]: Modify signature may because FE-BE incompatible when upgrading. So better not changing it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**